python/experiments.py

```diff
@@ -29,6 +30,8 @@ def task_compile_benchmarks():
     for benchmark in manifest["benchmarks"]:
         filepath = util.churchroad_evaluation_dir() / benchmark["filepath"]
         benchmark_name = filepath.stem
+        benchmark_flags = benchmark.get('flags')
+        features = benchmark.get('features')
```

Comment:
what happens if these fields aren't there? they should be optional. how does `get` react if e.g. `'flags'` isn't present?

Reply:
> what happens if these fields aren't there? they should be optional. how does `get` react if e.g. `'flags'` isn't present?

Ahh that's a good point. I think what would happen is that it would be assigned to `None`, but I've changed the lines to:
```
_benchmark_flags = benchmark.get("flags") if benchmark["flags"] else ""
_features = benchmark.get("features") if benchmark["features"] else {}
_extra_summary_fields = {"tool": "vivado", "name": benchmark_name}
    for feature in _features:
        _extra_summary_fields[feature] = _features.get(feature)
```

Would `features` best be implemented with a coarser API as you suggested for `flags`? 

Right now it's a dictionary provided by `manifest.yml` for each benchmark.